### DISCUSSION TITLE
Creating a binding in c# code

### DISCUSSION THREAD

**Discussion by pvlzh** [AvaloniaUI/Avalonia#8361] [Q&A]
Hello. I started to get acquainted with Avalonia, but I already have some experience in xaml. Can you explain please
Why does it fail with "Name scope is null or was already collected" error in Bind method?

> var binding = new Binding
> {
>     ElementName = "source", // TextBox x:Name="source"
>     Path = "Text", // TextBox.Text
> };
> // TextBlock x:Name="target"
> target.Bind(TextBlock.TextProperty, binding);

Although in xaml the binding works fine.
`
<TextBlock x:Name="target" Text="{Binding #source.Text}"/>
<TextBox x:Name="source"/>
`

**Comment by timunie** [comment]
Maybe this doc helps you: https://docs.avaloniaui.net/docs/data-binding/binding-from-code

Happy coding
Tim
